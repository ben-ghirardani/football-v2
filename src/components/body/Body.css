tr {
  color: black;
  border-bottom: 1px dotted darkgray;
}

tr:hover td {
  background-color: rgb(153, 255, 51);
}

.right-table {
  border-bottom-right-radius: 15px;
  border-top-right-radius: 15px;
}

.left-table {
  padding: 3px;
  border-bottom-left-radius: 15px;
  border-top-left-radius: 15px;
  text-align: center;
}