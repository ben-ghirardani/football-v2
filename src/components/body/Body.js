import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import './Body.css';

// sits inside Table with TableHeader
// accepts a single row of table data. 
class Body extends Component {

  constructor(props) {
    super(props);
    this.goToFixtures = this.goToFixtures.bind(this);
    this.teamNameEditor = this.teamNameEditor.bind(this);
  }

  // using react-router-dom, import withRouter, wrap the export of class in withRouter, 
  // can then call .push on the history object
  // https://stackoverflow.com/questions/42701129/how-to-push-to-history-in-react-router-v4

  // *** applies to v5? ***

  goToFixtures () {
    let team = this.props.team
    let id = this.props.id
    console.log(team, id)
    this.props.updateCurrentTeam(team, id);
    this.props.history.push("/fixtures");
  }

  teamNameEditor (name) {
    let firstEdit = name.replace("AFC ", "");
    let secondEdit = firstEdit.replace(" FC", "");
    return secondEdit;
  }

  render() {
        return(
          <tr
            onClick={this.goToFixtures}
          >
            <td>{this.props.position}</td>
            <td>{this.teamNameEditor(this.props.team)}</td>
            <td>{this.props.played}</td>
            <td>{this.props.won}</td>
            <td>{this.props.draw}</td>
            <td>{this.props.lost}</td>
            <td>{this.props.gf}</td>
            <td>{this.props.ga}</td>
            <td>{this.props.gd}</td>
            <td>{this.props.points}</td>
          </tr> 
        )
    }

}

export default withRouter(Body);