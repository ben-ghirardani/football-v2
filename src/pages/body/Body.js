import React, { Component } from 'react';
import './body.css'

// sits inside Table with Header
// accepts a single row of table data. 
export default class Body extends Component {

  render() {
        return(
            <tr>
                <td>{this.props.position}</td>
                <td>{this.props.team}</td>
                <td>{this.props.played}</td>
                <td>{this.props.won}</td>
                <td>{this.props.drawn}</td>
                <td>{this.props.lost}</td>
                <td>{this.props.gf}</td>
                <td>{this.props.ga}</td>
                <td>{this.props.gd}</td>
                <td>{this.props.points}</td>
            </tr>
        )
    }

}