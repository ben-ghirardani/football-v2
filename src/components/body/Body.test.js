import React from 'react';
import Body from './Body.js';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

let props;

beforeEach(() => {
  props = {
    position: 1,
    team: "Arsenal",
    id: 456,
    played: 10,
    won: 5,
    draw: 3,
    lost: 2,
    gf: 20,
    ga: 10,
    gd: 10,
    points: 18,
    teamNameEditor: jest.fn().mockReturnValue("Arsenal")
  }
})

describe('Body tests', () => {

  it('matches snapshot', ()=> {
    const wrapper = shallow(<Body {...props} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('renders a table row', ()=> {
    const wrapper = shallow(<Body {...props} />)
    expect(wrapper.find('tr')).toHaveLength(1);
  })

});