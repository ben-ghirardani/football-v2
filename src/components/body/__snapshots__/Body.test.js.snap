// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body tests Body renders as expected 1`] = `ShallowWrapper {}`;

exports[`Body tests matches snapshot 1`] = `ShallowWrapper {}`;
